AlpsSDK/Repositories/PubSub/PublicationRepository.swift

```diff
@@ -11,17 +11,27 @@ import Alps
 
 let kPublicationFile = "kPublicationFile.Alps"
 
-final public class PublicationRepository: AsyncCreateable, AsyncReadable, AsyncDeleteable {
+final public class PublicationRepository: AsyncCreateable, AsyncReadable, AsyncDeleteable, AsyncClearable {
+    
     typealias DataType = Publication
     
-    private(set) var items = [Publication]() {
+    private(set) var items: [Publication] {
+        get {
+            return _items.withoutExpired
+        }
+        set {
+            _items = newValue
+        }
+    }
+    
+    private var _items = [Publication]() {
         didSet {
-            _ = PersistenceManager.save(object: self.items.map { $0.encodablePublication }, to: kPublicationFile)
+            _ = PersistenceManager.save(object: self._items.withoutExpired.map { $0.encodablePublication }, to: kPublicationFile)
```

Comment:
It is good that we also keep the item that are expired, there might be some interesting use-case. 
I like your implementation 👍